From 1e08ea66bcca6d1444740a700bf6099b3ebc52c1 Mon Sep 17 00:00:00 2001
From: Anton Bambura <jenneron@protonmail.com>
Date: Fri, 29 Jul 2022 03:25:26 +0300
Subject: [PATCH 26/33] ARM: dts: exynos: snow: add usb hub

This device has a USB Hub which camera and modem are connected to.
Add support for it.

Signed-off-by: Anton Bambura <jenneron@protonmail.com>
---
 arch/arm/boot/dts/exynos5250-snow-common.dtsi | 23 +++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/arch/arm/boot/dts/exynos5250-snow-common.dtsi b/arch/arm/boot/dts/exynos5250-snow-common.dtsi
index 817b48c924f0..4ea63c44d86e 100644
--- a/arch/arm/boot/dts/exynos5250-snow-common.dtsi
+++ b/arch/arm/boot/dts/exynos5250-snow-common.dtsi
@@ -81,6 +81,22 @@ i2c_104: i2c@0 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
+			usb-hub@8 {
+				compatible = "smsc,usb3503a";
+				reg = <0x08>;
+
+				reset-gpios = <&gpe1 0 GPIO_ACTIVE_LOW>;
+				disabled-ports = <3>;
+				initial-mode = <1>;
+
+				clock-names = "refclk";
+				clocks = <&pmu_system_controller 0>;
+				refclk-frequency = <24000000>;
+
+				pinctrl-names = "default";
+				pinctrl-0 = <&usb_hub_reset>;
+			};
+
 			battery: sbs-battery@b {
 				compatible = "sbs,sbs-battery";
 				reg = <0xb>;
@@ -704,6 +720,13 @@ arb_our_claim: arb-our-claim-pins {
 		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
 		samsung,pin-drv = <EXYNOS4_PIN_DRV_LV1>;
 	};
+
+	usb_hub_reset: usb-hub-reset-pins {
+		samsung,pins = "gpe1-0";
+		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+		samsung,pin-drv = <EXYNOS4_PIN_DRV_LV1>;
+	};
 };
 
 &pmu_system_controller {
-- 
2.38.0

